lib/verify.go

```diff
@@ -142,6 +152,23 @@ func VerifyChain(certs []*x509.Certificate, ocspStaple []byte, dnsName, caPath s
 		return result
 	}
 
+	if spiffeIDExpected {
+		// The Verify method above didn't actually verify that the certificate matches the expected
+		// SPIFFE ID. We thus perform this check explicitly here.
+		expectedSpiffeID := expectedName
+		matched := false
+		for _, uri := range certs[0].URIs {
```

Comment:
It's invalid for a SPIFFE cert to have more than one URI SAN. Should we warn in this case?

Reply:
I'm going with failing the match when there's not exactly one URI SAN, as per X.509 SPIFFE SVID spec.